algebra-utils/src/msm/variable_base.rs

```diff
@@ -176,11 +380,50 @@ impl VariableBaseMSM {
         G: AffineCurve,
         G::Projective: ProjectiveCurve<Affine = G>
     {
-        #[cfg(not(feature = "msm_affine"))]
-        return Self::multi_scalar_mul_mixed(bases, scalars);
 
-        #[cfg(feature = "msm_affine")]
-        return Self::multi_scalar_mul_affine(bases, scalars);   
+        let scal_len = scalars.len();
+
+        if TypeId::of::<G>() == TypeId::of::<algebra::curves::bn_382::G1Affine>()
+        {
+            let c: usize = if scal_len < 32 {
+                3
+            } else if (scal_len < 1 << 17) || (scal_len > 1 << 23) {
+                (2.0 / 3.0 * (f64::from(scalars.len() as u32)).log2() - 2.0).ceil() as usize
+            } else if scal_len < 1 << 21 {
+                12
+            } else {
+                16
+            };
+            return Self::multi_scalar_mul_affine_sd_c(bases, scalars, c);
+        } else if TypeId::of::<G>() == TypeId::of::<algebra::curves::tweedle::dee::Affine>() {
```

Comment:
This if should be enable iff the feature "tweedle" is enabled. Transform this else if into a simple if and add #[cfg(feature = "tweedle")] between line 397 and 398